docs/data/material/guides/typescript/typescript.md

```diff
@@ -49,4 +49,21 @@ Moved to [/customization/theming/#custom-variables](/material-ui/customization/t
 
 ## Usage of `component` prop
 
-Moved to [/guides/composition/#with-typescript](/material-ui/guides/composition/#with-typescript).
+The usage of the `component` prop can be problematic if you are creating your custom component based on the Mateiral UI's components.
```

Comment:
```suggestion
Using the `component` prop can be problematic if you are creating your custom component based on Material UI's components.
```

Reply:
Should we elaborate on _why_ it's a problem?

Reply:
Because of TypeScript limitations. Maybe we should rephrase to:

```suggestion
Because of some TypeScript limitations, using the `component` prop can be problematic if you are creating your custom component based on the Material UI's components.
```

Reply:
I think that's worth mentioning for clarity. 👍